pylint/checkers/typecheck.py

```diff
@@ -1720,9 +1720,18 @@ def visit_subscript(self, node):
             return
 
         inferred = safe_infer(node.value)
+
         if inferred is None or inferred is astroid.Uninferable:
             return
 
+        if inferred.decorators:
```

Comment:
I think we should enter in these lines only if the protocol is not supported for the current inferred node.

Reply:
I initially did that way because a node could support the protocol, but its decorator remove it.

Dumb example:

```python
def deco(f):
    def wrapped():
        return "Hello"
    return wrapped
```

So I'd prefer to only trust the last decorator of the node than the node itself. But It's my first commit on pylint and first time I use astropy, so if you're sure, I'll gladly do as you feel :)